From de919fc1cf4250db240757a94f302d81620b451c Mon Sep 17 00:00:00 2001
From: Haihua Hu <b55597@freescale.com>
Date: Tue, 8 Dec 2015 14:09:58 +0800
Subject: [PATCH 5/5] [MMFMWK-6966] [glplugin] fix gleffects fisheye shader
 compile error

Shader function "sqrt" is not supported on some embedded platform, having "sqrt(2)"
in the shader does not actually gain anybody anything. Use float number 1.41421.

https://bugzilla.gnome.org/show_bug.cgi?id=761271
Upstream-Status: Backport [1.6.4]

Signed-off-by: Haihua Hu <b55597@freescale.com>
---
 ext/gl/effects/gstgleffectssources.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/ext/gl/effects/gstgleffectssources.c b/ext/gl/effects/gstgleffectssources.c
index 3da856c..076e21b 100644
--- a/ext/gl/effects/gstgleffectssources.c
+++ b/ext/gl/effects/gstgleffectssources.c
@@ -152,7 +152,7 @@ const gchar *fisheye_fragment_source_gles2 =
   "  vec2 normcoord;"
   "  normcoord = texturecoord - 0.5;"
   "  float r = length (normcoord);"
-  "  normcoord *= r * sqrt(2);"
+  "  normcoord *= r * 1.41421;" /* sqrt (2) */
   "  texturecoord = normcoord + 0.5;"
   "  gl_FragColor = texture2D (tex, texturecoord);"
   "}";
-- 
1.7.9.5

